### PR TITLE
Refactored Calculator.js by implementing hooks

### PR DESCRIPTION
**Pull request by StarSheriff2**
Hi Code Reviewer 😃 ,

In this feature branch I did the following:

- Refactored all components from class based components to functional components with hooks
- Used try/catch to handle error in Calculator.js
- Use useEffect to handle errors

### CODE REVIEW

File: src/components/otherOperations.js [hunk 1]
```diff
@@ -1,29 +1,23 @@
...
-      </div>
-    );
-  }
-}
+  return (
+    <div className="Other-Operations-container">
+      <button onClick={handleClick} type="button" className="Other-Operations-btn btn" value="AC">AC</button>
+      <button onClick={handleClick} type="button" className="Other-Operations-btn btn" value="+/-">+ / -</button>
+      <button onClick={handleClick} type="button" className="Other-Operations-btn btn" value="%">%</button>
+    </div>
```
Comment (DeliceLydia) [inline src/components/otherOperations.js:15]: - [x] Please work on the functionality of `%` button currently its not working

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Hi @StarSheriff2👏,

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/otherOperations.js (hunk 1 above)
Comment (StarSheriff2) [reply src/components/otherOperations.js:15]: Hi Delice,

I tried this button, and it works as expected. Bear in mind that it's not a percentage button but a modulus operator. Its functionality was determined by the "calculate" and "operate" files provided to us by Micorverse.

Comment (StarSheriff2) [reply src/components/otherOperations.js:15]: Here's a link with a video showing the modulus button working as expected: [https://drive.google.com/file/d/1XumsxbwelpLsbwPSfOMZtlbkk2ew09Xq/view?usp=sharing](https://drive.google.com/file/d/1XumsxbwelpLsbwPSfOMZtlbkk2ew09Xq/view?usp=sharing)

**Pull request merged**